board_state.go

```diff
@@ -0,0 +1,19 @@
+package rules
+
+func (b *BoardState) Copy() *BoardState {
+	nextState := &BoardState{
+		Height:  b.Height,
+		Width:   b.Width,
+		Food:    append([]Point{}, b.Food...),
+		Snakes:  make([]Snake, len(b.Snakes)),
+		Hazards: append([]Point{}, b.Hazards...),
+	}
+	for i := 0; i < len(b.Snakes); i++ {
+		nextState.Snakes[i].ID = b.Snakes[i].ID
+		nextState.Snakes[i].Health = b.Snakes[i].Health
+		nextState.Snakes[i].Body = append([]Point{}, b.Snakes[i].Body...)
+		nextState.Snakes[i].EliminatedCause = b.Snakes[i].EliminatedCause
+		nextState.Snakes[i].EliminatedBy = b.Snakes[i].EliminatedBy
+	}
+	return nextState
+}
```

Comment:
I made the exact same change in the refactor I just merged 😆 
The only difference is the name - there's a `Clone` method on BoardState already now.

Reply:
great minds think alike :p